src/rpc_client/mod.rs

```diff
@@ -29,7 +31,14 @@ impl RpcContext {
 
 #[async_trait]
 pub trait RpcClient: Send + Sync {
-    async fn query(&self, ctx: &RpcContext, req: &QueryRequestPb) -> Result<QueryResponsePb>;
-    async fn write(&self, ctx: &RpcContext, req: &WriteRequestPb) -> Result<WriteResponsePb>;
-    async fn route(&self, ctx: &RpcContext, req: &RouteRequestPb) -> Result<RouteResponsePb>;
+    async fn query(&self, ctx: &RpcContext, req: QueryRequestPb) -> Result<QueryResponsePb>;
+    async fn write(&self, ctx: &RpcContext, req: WriteRequestPb) -> Result<WriteResponsePb>;
+    async fn route(&self, ctx: &RpcContext, req: RouteRequestPb) -> Result<RouteResponsePb>;
+}
+
+#[async_trait]
+pub trait RpcClientFactory: Send + Sync {
+    // The Build method may fail because of invalid endpoint, so it returns a
+    // Result. Any caller calls this method should handle the potencial error
```

Comment:
```suggestion
    /// Build `RpcClient`.
    ///
    /// It may fail because of invalid endpoint,  any caller calls this method should handle the potential error.
```